File: Example/javaapp/src/main/java/com/stripe/example/javaapp/fragment/offline/OfflinePaymentsLogHolder.java
package com.stripe.example.javaapp.fragment.offline;

import android.view.View;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.stripe.example.javaapp.R;
import com.stripe.example.javaapp.model.Event;
import com.stripe.example.javaapp.model.OfflineLog;

import org.jetbrains.annotations.NotNull;

/**
 * A simple [RecyclerView.ViewHolder] that displays various offline payments logs
 */
public class OfflinePaymentsLogHolder extends RecyclerView.ViewHolder {

    public OfflinePaymentsLogHolder(@NotNull View itemView) {
        super(itemView);
    }

    public void bind(@NotNull OfflineLog log) {
        ((TextView) itemView.findViewById(R.id.message)).setText(log.toMessage());
    }
}

File: Example/kotlinapp/src/main/java/com/stripe/example/fragment/offline/OfflinePaymentsLogHolder.kt
package com.stripe.example.fragment.offline

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.stripe.example.R
import com.stripe.example.model.OfflineEvent

/**
 * A simple [RecyclerView.ViewHolder] that displays various offline payments logs
 */
class OfflinePaymentsLogHolder(
        itemView: View
) : RecyclerView.ViewHolder(itemView) {
    fun bind(event: OfflineEvent) {
        (itemView.findViewById<View>(R.id.message) as TextView).text = event.summary
    }
}
